src/main/java/muramasa/antimatter/cover/Cover.java

```diff
@@ -146,8 +149,11 @@ public static ModelResourceLocation getBasicModel() {
     }
 
     public void onRegister() {
-
+        String id = tier == null ? getId() : (getId() + "_" + tier.getId());
+        if (AntimatterAPI.get(Cover.class, id) == null)
+            AntimatterAPI.register(Cover.class, id, this);
     }
+
```

Comment:
Take a look at AntimatterAPI#register(IAntimatterObject)

Reply:
This is a workaround because covers are stateful and may be created many times. If/when cover will be a singleton no checks will be necessary

Reply:
Understood, we'll get to that later then.